packages/service-library/src/servicelib/docker_utils.py

```diff
@@ -281,3 +284,23 @@ async def pull_image(
                 f"pulling {image_short_name}: {pull_progress}...",
                 logging.DEBUG,
             )
+
+
+async def create_remote_docker_client(
+    settings: DockerApiProxysettings,
+) -> aiodocker.Docker:
+    """Allows to attach to a remote docker client istnace and use it in place
+    of the docker socket.
+
+    NOTE: Usually this is required to use a manager node's docker client for swarm commands.
+    """
+    session: ClientSession | None = None
+    if settings.DOCKER_API_PROXY_USER and settings.DOCKER_API_PROXY_PASSWORD:
+        session = ClientSession(
+            auth=aiohttp.BasicAuth(
+                login=settings.DOCKER_API_PROXY_USER,
+                password=settings.DOCKER_API_PROXY_PASSWORD.get_secret_value(),
+            )
+        )
+
+    return aiodocker.Docker(url=settings.base_url, session=session)
```

Comment:
aiodocker.Docker is usually used as a context manager. who is calling the client close() call here?
you might want to create a context manager as well

Reply:
I was wondering about that.

Q: how do you implement the setup/teardown pattern with a context manager? This is what blocked me from having it.

Reply:
OK I've managed to make something nice. Which will also be useful in the future. I have transformed the setup of the client in a lifespan context manager.
I have also added some utils to combine multiple lifespan context managers.

Reply:
Now the integration tests actually use and application build using these new features.